Fix-Raiden-Boss 2.0 (for all user )/RaidenBossFix2.py

```diff
@@ -24,85 +26,541 @@
-for k in dict(parser).keys():
-    if pattern.match(k) != None:
-        blend = dict(parser[k])
-if blend["draw"]:
-    draw: str = blend["draw"]
-
-fixedBlendName = blendCorrection().split('.')[0]
-
-addFix = f'\n;raiden boss fixed by NK#1321 if you used it for fix your raiden pls give credit for "Nhok0169"\n;thank nguen#2011 SilentNightSound#7430 and HazrateGolabi#1364 for support\n\n[TextureOverride{fixedBlendName}]\nhash = fe5c0180\nvb1 = Resource{fixedBlendName}\nhandling = skip\ndraw = {draw} \n\n[Resource{fixedBlendName}]\ntype = Buffer\nstride = 32\nfilename = {fixedBlendName}.buf\n\n\n'
-
-original = ''
-with open(ini_file, "r") as f:
-    original = f.read()
-dis_ini(ini_file)
-
-# writing the fixed file
-print("making the fixed ini file")
-with open(ini_file, "w") as f:
-    f.write(addFix)
-    f.write(original)
-
-
-print("ENJOY")
\ No newline at end of file
+MaxVGIndex = 117
+
+DefaultFileType = "file"
+DefaultPath = "."
+IniExt = ".ini"
+MergedFile = f"merged{IniExt}"
+
+IniFileType = "*.ini file"
+BlendFileType = "Blend.buf"
+
+
+class Error(Exception):
+    def __init__(self, message: str):
+        super().__init__(f"ERROR: {message}")
+
+
+class FileException(Error):
+    def __init__(self, message: str, path: str = DefaultPath):
+        if (path != DefaultPath):
+            message += f" at {path}"
+
+        super().__init__(message)
+
+
+class DuplicateFileException(FileException):
+    def __init__(self, fileType: str = DefaultFileType, path: str = DefaultPath):
+        message = f"Ensure only one {fileType} exists"
+        super().__init__(message, path = path)
+
+
+class MissingFileException(FileException):
+    def __init__(self, fileType: str = DefaultFileType, path: str = DefaultPath):
+        message = f"Unable to find {fileType}. Ensure it is in the folder"
+        super().__init__(message, path = path)
+
+
+class BlendFileNotRecognized(FileException):
+    def __init__(self, blendFile: str):
+        super().__init__(f"Blend file format not recognized for {os.path.basename(blendFile)}", path = os.path.dirname(blendFile))
+
+
+class FileService():
+    @classmethod
+    def getFilesAndDirs(cls, path: str = DefaultPath) -> List[List[str]]:
+        files = []
+        dirs = []
+
+        pathItems = os.listdir(path)
+
+        for itemPath in pathItems:
+            fullPath = os.path.join(path, itemPath)
+            if (os.path.isfile(fullPath)):
+                files.append(fullPath)
+            else:
+                dirs.append(fullPath)
+
+        return [files, dirs]
+
+    @classmethod
+    def isFile(cls, path: str, file: str) -> bool:
+        fullPath = os.path.join(path, file)
+        return os.path.isfile(fullPath)
+
+    # filters and partitions the files based on the different filters specified
+    @classmethod
+    def getFiles(cls, path: str = DefaultPath, filters: List[Callable[[str], bool]] = [], files: Optional[List[str]] = None) -> Union[List[str], List[List[str]]]:
```

Comment:
Filters different types of files (eg. blend or ini files) in a single pass rather than multiple passes